include/fiction/io/write_sqd_layout.hpp

```diff
@@ -254,11 +255,32 @@
                 if constexpr (has_sidb_technology_v<Lyt>)
                 {
                     const auto type = this->lyt.get_cell_type(c);
-                    const auto type_str =
-                        type == sidb_technology::cell_type::NORMAL ? "" :
-                        type == sidb_technology::cell_type::INPUT  ? fmt::format(siqad::DOT_TYPE, "input") :
-                        type == sidb_technology::cell_type::OUTPUT ? fmt::format(siqad::DOT_TYPE, "output") :
-                                                                     "";
+
+                    std::string type_str;
+
+                    switch (type)
+                    {
+                        case (sidb_technology::cell_type::NORMAL):
+                        {
+                            type_str = "";
+                            break;
+                        }
+                        case (sidb_technology::cell_type::INPUT):
+                        {
+                            type_str = fmt::format(siqad::DOT_TYPE, "input");
+                            break;
+                        }
+                        case (sidb_technology::cell_type::OUTPUT):
+                        {
+                            type_str = fmt::format(siqad::DOT_TYPE, "output");
+                            break;
+                        }
+                        case (sidb_technology::cell_type::LOGIC):
+                        {
+                            type_str = fmt::format(siqad::DOT_TYPE, "logic");
+                            break;
+                        }
+                    }
```

Comment:
This is a dilemma. It is true that the empty cell-type exists. However, it can never occur because it is not possible to add it to the cell-level layout. Therefore, it is impossible to test this case in the unit test and the test coverage fails. How do we handle this, @marcelwa?

Reply:
In such cases, you could add the missing cases or add a default case (I would prefer the explicit listing of cases, though), and then exclude that block from coverage analysis via the magic comments:
```c++
// LCOV_EXCL_START
                        case (sidb_technology::cell_type::EMPTY):
                        {
                            // this case can never happen; it exists to comfort the compilers
                            break;
                        }                       
// LCOV_EXCL_STOP